modelator/utils/apalache_helpers.py

```diff
@@ -24,6 +26,11 @@ def write_trace_files_to(apalache_result: Dict, traces_dir: str) -> List[str]:
         f for f in apalache_result["files"].keys() if itfs_filenames_pattern.search(f)
     ]
     itfs_filenames.sort()
+    if simulate is True and len(itfs_filenames) > 0:
+        # have to filter out the "example0.itf.json" because in the simulation mode and older versions
+        # of Apalache (e.g., current Modelator's default, 0.25.10), an additional examples under the name
+        # "example0.itf.json" is generated.
+        itfs_filenames = [f for f in itfs_filenames if not f == "example0.itf.json"]
```

Comment:
1. You can just use `!=`.
2. I would refrain from using `"example0.itf.json"`. Maybe `not re.search("[^\d]0+.itf.json$", f)`.

Reply:
Why 2. ? I assume you are not sure about Apalache always calling them examples. But I think it is overoptimization: they could also change the numbering (e.g., start from 0 instead of 1), or they could add any other kind of changes to the name (e.g., `0_example.itf.json`). Thus, I am more inclined to keep it as is (but open if you insist on the suggested change for a good reason).

Reply:
Ok. Makes sens. Anyway, this is a hack. So it should look like a hack :smile: lol